frontend/src/components/managePermissions/ManagePermissions.vue

```diff
@@ -19,16 +19,17 @@ import {
 } from '@/store/NotificationState';
 import { Severity } from '@/enum/SeverityEnum';
 import type { FamApplicationUserRoleAssignmentGet } from 'fam-app-acsctl-api';
+import { requireInjection } from '@/services/utils';
 
-const apiServiceFactory = new ApiServiceFactory();
-const applicationsApi = apiServiceFactory.getApplicationApi();
-const userRoleAssignmentApi = apiServiceFactory.getUserRoleAssignmentApi();
+const apiService = requireInjection(ApiServiceFactory.SERVICE_KEY); // Inject service
 const userRoleAssignments = shallowRef<FamApplicationUserRoleAssignmentGet[]>();
 
 onMounted(async () => {
     // Reload list each time we navigate to this page to avoid forcing user to refresh if their access changes.
     applicationsUserAdministers.value = (
-        await applicationsApi.getApplications()
+        await apiService
+            .getAppAccessControlApiService()
```

Comment:
I wonder if we still want to create a variable for these apis first? instead of every time call the whole chain "apiService.getAppAccessControlApiService()....."? maybe still can have:

const applicationsApi = apiService.getAppAccessControlApiService().applicationsApi
const userRoleAssignmentApi = apiService.getAppAccessControlApiService().userRoleAssignmentApi
etc.?

Reply:
Yes, I have noticed that as well, but didn't put much thought of it yet.
For now for long chain might be ok within the component itself (like you did) and can find an idea later.
However, I have been thinking since long time ago to get rid of this `onMounted` if necessary, so some magic might be good to happen on the routing level.
Or..., maybe create separate key(s) for the two `applicationsApi` and `forestClientsApi` as like what I did for the `apiService`. Let me try...

Reply:
Hi @MCatherine1994,
I have now provide two different services("appAccessControlApiService" and "adminManagementApiService") for injection available for convenience.
It will be up to the component to decide what to call from the service within the component.
I assume that would not break the test mocking.